rest/adminapitest/admin_api_test.go

```diff
@@ -2141,6 +2141,73 @@ func TestHandleCreateDB(t *testing.T) {
 	rest.RequireStatus(t, resp, http.StatusBadRequest)
 }
 
+func TestHandleCreateDBJsonName(t *testing.T) {
+	if base.UnitTestUrlIsWalrus() {
+		t.Skip("This test only works against Couchbase Server")
+	}
+	testCases := []struct {
+		name        string
+		JSONname    string
+		expectError bool
+	}{
+		{
+			name:        "Name match",
+			JSONname:    "db1",
+			expectError: false,
+		},
+		{
+			name:        "Name mismatch",
+			JSONname:    "dummy",
+			expectError: true,
+		},
+		{
+			name:        "No JSON Name",
+			JSONname:    "",
+			expectError: false,
+		},
+	}
+
+	for _, test := range testCases {
+		t.Run(test.name, func(t *testing.T) {
+			tb := base.GetTestBucket(t)
+			defer tb.Close()
+
+			rt := rest.NewRestTester(t, &rest.RestTesterConfig{
+				CustomTestBucket: tb,
+				PersistentConfig: true,
+				MutateStartupConfig: func(config *rest.StartupConfig) {
+					config.Bootstrap.ConfigUpdateFrequency = base.NewConfigDuration(0)
+				},
+			})
+			defer rt.Close()
+
+			var resp *rest.TestResponse
+			if test.name != "No JSON Name" {
```

Comment:
Using the test name to decide what action to take seems a bit wrong to me as it means changing the test in the future may cause unexpected issues. It would be better here to instead use the if statement to check if `test.JSONname` is not empty and then continue with the logic to insert that name in the database config.